### PR TITLE
[RISCV] Add getFeaturesForCPU function support

### PR DESCRIPTION
**Pull request by 4vtomat**
This function parse the cpu and return it's supported
features placed in EnabledFeatures. It is same as the
one in X86TargetParser and also is used in IREE.

### CODE REVIEW

File: llvm/lib/TargetParser/RISCVTargetParser.cpp [hunk 1]
```diff
@@ -95,5 +96,31 @@ void fillValidTuneCPUArchList(SmallVectorImpl<StringRef> &Values, bool IsRV64) {
 #include "llvm/TargetParser/RISCVTargetParserDef.inc"
 }
 
+void getFeaturesForCPU(StringRef CPU,
```
Comment (topperc) [inline llvm/lib/TargetParser/RISCVTargetParser.cpp:99]: This code is dead in tree. We should have a comment indicating why it shouldn't be deleted.

File: llvm/lib/TargetParser/RISCVTargetParser.cpp [hunk 2]
```diff
@@ -95,5 +96,31 @@ void fillValidTuneCPUArchList(SmallVectorImpl<StringRef> &Values, bool IsRV64) {
...
+  auto RII = RISCVISAInfo::parseArchString(
+      MarchFromCPU, /* EnableExperimentalExtension */ true);
+
+  if (llvm::errorToBool(RII.takeError()))
+    return;
+
+  std::vector<std::string> FeatStrings =
+      (*RII)->toFeatures(/* AddAllExtensions */ true);
+  for (const auto &F : FeatStrings) {
+    if (F[0] == '-')
```
Comment (topperc) [inline llvm/lib/TargetParser/RISCVTargetParser.cpp:116]: Why pass AddAllExtensions=true if you're going to skip the '-' ones?

File: llvm/lib/TargetParser/RISCVTargetParser.cpp (hunk 1 above)
Comment (4vtomat) [reply llvm/lib/TargetParser/RISCVTargetParser.cpp:99]: Good point, thanks!

File: llvm/lib/TargetParser/RISCVTargetParser.cpp [hunk 3]
```diff
@@ -95,5 +96,28 @@ void fillValidTuneCPUArchList(SmallVectorImpl<StringRef> &Values, bool IsRV64) {
...
+  if (llvm::errorToBool(RII.takeError()))
+    return;
+
+  std::vector<std::string> FeatStrings =
+      (*RII)->toFeatures(/* AddAllExtensions */ false);
+  for (const auto &F : FeatStrings)
+    if (NeedPlus)
+      EnabledFeatures.push_back(F);
+    else
+      EnabledFeatures.push_back(F.substr(1, F.size() - 1));
```
Comment (topperc) [inline llvm/lib/TargetParser/RISCVTargetParser.cpp:120]: I think we can use `F.substr(1)`?

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**